packages/terra-clinical-onset-picker/src/OnsetPicker.jsx

```diff
@@ -130,9 +130,57 @@ class OnsetPicker extends React.Component {
       onsetDate: this.props.onsetDate ? moment(this.props.onsetDate) : undefined,
       age: ageValues.age,
       ageUnit: ageValues.ageUnit,
+      year: undefined,
+      month: undefined,
     };
   }
 
+  componentDidMount() {
+    if (this.props.onsetDate) {
+      if (this.props.granularity === 'year' && this.props.onsetDate === moment().year(moment(this.props.onsetDate).year()).startOf('year').format(DATE_FORMAT)) {
```

Comment:
Please add appropriate code comments or introduce a meaningful variable for `moment().year(moment(this.props.onsetDate).year()).startOf('year').format(DATE_FORMAT))`for better readability.

Reply:
Added variables in 1417f94